### DISCUSSION TITLE
cannot put x-overflow item into DLX?

### DISCUSSION THREAD

**Discussion by amills-vibeirl** [rabbitmq/rabbitmq-server#10361] [Other]
I am trying to automatically put items that exceed x-max-length into a DLX.
Is it possible to do this with RMQ?
https://stackoverflow.com/questions/77804007/when-using-x-max-length-and-x-overflow-how-to-send-to-dead-letter-exchange

**Comment by michaelklishin** [comment]
[Queue Overflow Behaviour](https://rabbitmq.com/maxlength.html#overflow-behaviour) lists all the available options.

`drop-head` is very specific about what it does: it **drops** the excess. You cannot use it and expect messages to be dead-lettered at the same time. Perhaps try `reject-publish-dlx`.
